src/satosa/frontends/openid_connect.py

```diff
@@ -78,7 +78,14 @@ def _create_provider(self, endpoint_baseurl):
 
         authz_state = self._init_authorization_state()
         db_uri = self.config.get("db_uri")
-        cdb = MongoWrapper(db_uri, "satosa", "clients") if db_uri else {}
+        cdb_file = self.config.get("db_file")
```

Comment:
Is this line needed? I don't see where cdb_file is set.

Shouldn't the following be enough?
```python
if db_uri:
    cdb = MongoWrapper(db_uri, "satosa", "clients")
elif client_db_path:
    with open(self.config['client_db_path']) as f:
        cdb = json.loads(f.read())
else:
   cdb = {}
```